functions/Install-DbaMaintenanceSolution.ps1

```diff
@@ -44,6 +44,12 @@ function Install-DbaMaintenanceSolution {
         Specifies the path to a local file to install Ola's solution from. This *should* be the zip file as distributed by the maintainers.
         If this parameter is not specified, the latest version will be downloaded and installed from https://github.com/olahallengren/sql-server-maintenance-solution
 
+    .PARAMETER NoDataLoss
+        Normally, when the ReplaceExisting flag is used, it will drop the
+        CommandLog table and re-create it. When this flag is also passed, it
+        will instead rename the old table with a time stamp which will allow
+        you to do a migration of existing data to the new table.
+
```

Comment:
I'm probably in the boat by myself but I don't think this belongs in the command at all.

Reply:
We have `Copy-DbaDbTableData` or other command that someone can do this via their own wrapper if they really want that data. The functionality of the install command should be scoped to only deal with installation....not saving data.

Reply:
In that case the removal of the commandlog table belongs here even less.
As part of our service we add some functionality. It's important to either offer a water tight combination of choices or to leave the extras to the user. I find the current options somewhat incongruent.

Reply:
So you expect this command to simply execute the SQL files provided by Ola?

Reply:
> I think this is a great idea, but find the parameter name confusing.
> Maybe `-BackupCommandLog` ?

I'm happy to call it whatever. I picked a name that made sense to me. Is `-BackupCommandLog` your recommendation?

Reply:
> Just noticed the jobs removed are all that have 'hallengren' in the description!?!? That is a very dangerous assumption! Shouldn't we fix that as we're working on it now?

Perhaps. But I'd counter with "What are the odds that another job has the word 'hallengren' in the description?". IMO, it's a fairly good discriminator.

Reply:
too high!
maybe not for you or me, but in Denmark or Sweden it is more likely.
I see two options:
- we provide the list of jobs
- we find jobs that have a jobstep in which they execute one of the maintenancesolution stored procedures

Reply:
> 
> 
> So you expect this command to simply execute the SQL files provided by Ola?

@wsmelton certainly not, enrichment is what we live for 😄 

But if we add functionality around 'working with the command log table' it should be a complete set and not just drop it.

Reply:
Oh, I don't want this parameter in the command at all. This is for pure installation purposes, not taking backups and leaving them hanging around for the DBA to deal with months down the road with no support to clean it up....which is primary reason I see taking a backup of that table upon the user's responsibility and not in this command (or our module itself).

Reply:
The takeaway from the conversation we had about this on slack was that the user assumes responsibility for reading the documentation and acting in accordance with their own needs/best interests. To that end, I assumed that by providing the parameter that the user has read and understood the implications and has a plan to deal with the old data.

Reply:
That was part of the conversation.

Reply:
I don't see a need for it nor do I think it belongs in this command. If the issue as a whole is that we are overwriting or dropping objects, then simply remove the parameter to replace existing....that in all is not even required except for the table object. The scripts that Ola provides are going to alter the procedures anyway.

Reply:
Except that removing the `-ReplaceExisting` parameter precludes using this command for upgrading an existing installation of the scripts. If I'm understanding you correctly, the options to support that use case are:

1. Use this script but accept that it's going to drop the table.
2. Don't use this script.

Reply:
Any script can just be run as-is with no validating that the objects currently exists. If the procedure already exists, they will be updated because his scripts are [using ALTER PROCEDURE](https://github.com/olahallengren/sql-server-maintenance-solution/blob/master/IndexOptimize.sql). The script for the CommandLog is doing an if check so if it already exists it will just exist.

So to give users awareness of that if the table already exists a warning should be thrown to let them know that, but the replace existing parameter is not required at all and should be removed.

If you want to "update" that table with the latest script, then you take care of removing the current table object on your own, outside of this command. You have a plethora of options to chose from with our module to accomplish that across your whole enterprise. Whether you want to "save the data" or just drop the object:

1. Use `Copy-DbaDbTableData` to copy the data to a new table, then drop the original table
1. Use `Write-DbaDbTableData` to just write it to another table, then drop the original table (the copy is a bit faster for large tables)
1. Use `Get-DbaDbTable` and call the method to just drop the table
1. Use `Invoke-DbaQuery` and call T-SQL statement to drop the table

Removing all logic for that ReplaceExisting will allow you to just run any of the scripts as-is. Even for the table command because it won't execute the CREATE TABLE unless the table doesn't exists.

Reply:
> The script for the CommandLog is doing an if check so if it already exists it will just exist.

100% agree. However, the command as written has taken an opinionated stance that what the script does is insufficient and instead drops the table. I was trying to work within the spirit of that. Which I interpreted to be "`Install-DbaMaintenanceSolution` wants the CommandLog table's schema to match what is provided by the script regardless of whether the table existed prior to command invocation or not". Do I have that incorrect?

> So to give users awareness of that if the table already exists a warning should be thrown to let them know that, but the replace existing parameter is not required at all and should be removed.

Are you saying that's how it _should_ work or how it _does_ work? If the former, I'd be happy to make that change as it seems to be more in line with Ola's intent. And, to be explicit, the proposed change would be to remove the check for table existence as a reason to abort the command.